src/index/IndexBuilderMain.cpp

```diff
@@ -152,6 +156,12 @@ int main(int argc, char** argv) {
     index.memoryLimitIndexBuilding() = stxxlMemory.value();
   }
 
+  // If no external vocabulary basename was specified, use the same as the
+  // index basename.
+  if (baseNameVocabulary.empty()) {
+    baseNameVocabulary = baseNameIndex;
+  }
```

Comment:
If the `baseNameVocabulary` is an `std::optional<string>` then you can pass it as is all the way down, and only in a single place in the index (in `setOnDiskBase`) you have to resolve the default (or not even there but only in the functions that interact with the vocabulary).